MAUI/DataForm/localization.md

```diff
@@ -0,0 +1,183 @@
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can also localize the DataFormItem `LabelText`, `PlaceHolderText`, `GroupName` in the `GenerateDataFormItem` event of SfDataForm by using the Resources (.Resx) file.
+
+Here, string member of .resx file will be accessed through the class (in resxFilename.Designer.cs) which was auto-generated when .resx file created and static string members get localized using [ResourceManager](https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager.getstring?view=net-7.0) based on culture.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Name = "First Name", GroupName = "Details", Prompt = "Type first name")]
+public String FirstName { get; set; }
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem?.GroupName == "Details")
+        {
+            e.DataFormItem.GroupName = DataFormLocalization.GroupName;
+        }
+        if (e.DataFormItem.LabelText == "First Name")
+        {
+            e.DataFormItem.LabelText = DataFormLocalization.FirstName;
+            e.DataFormItem.PlaceholderText = DataFormLocalization.PlaceHolder;
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Localizing validation error messages
+
+Here, the validation (`Required`,`StringLength`) attributes or data form error messages get localized based on culture from Localization Resource File (.Resx).
+
+#### Using attribute
+
+The `Required` and `StringLength` attributes error message can be localized using [ErrorMessageResourceType](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) and [ErrorMessageResourceName](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) properties which are used to get a localized error messages from Localization Resource File (.Resx) based on culture.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Required(ErrorMessage = "Value should not be empty", ErrorMessageResourceName = "ErrorMessage", ErrorMessageResourceType = typeof(DataFormLocalization))]
+    [StringLength(15, ErrorMessage = "Enter proper name", ErrorMessageResourceName ="ErrorMessageLength", ErrorMessageResourceType =typeof(DataFormLocalization))]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can also localize the data form error message in the ` ValidateProperty` event of SfDataForm by using the Resources (.Resx) file.
```

Comment:
You can also localize the data form error message and valid message in the ` ValidateProperty` event of `SfDataForm` by using the Resources (.Resx) file.

Reply:
content changed